youths/api.py

```diff
@@ -0,0 +1,55 @@
+import logging
+from datetime import date
+
+from django.utils.translation import ugettext_lazy as _
+from rest_framework import permissions, serializers, viewsets
+from rest_framework.exceptions import APIException
+from rest_framework.renderers import BrowsableAPIRenderer, JSONRenderer
+
+from youths.models import YouthProfile
+
+logger = logging.getLogger(__name__)
+
+
+class YouthProfileAlreadyExists(APIException):
+    status_code = 409
+    default_detail = _("The youth profile for this user already exists.")
+    default_code = "profile_already_exists"
+
+
+class YouthProfileSerializer(serializers.ModelSerializer):
+    class Meta:
+        exclude = ["id"]
+        depth = 1
+        model = YouthProfile
+
+    def create(self, validated_data):
+        # Membership always expires at the end of the season (30.6.).
+        # Signups before May expire in the summer of the same year, others next year.
+        if "expiration" in validated_data:
```

Comment:
should there also be some logic in the model's `save()` method, for the objects created through non-API methods? or maybe we could pass a function to the `default=` attribute of the `expiration` field on the model...

Reply:
Good point, I guess these could also created with non-API methods... I'll move this logic to the `default` of the model field.